frontend/src/app/Settings/Users/ManageUsers.tsx

```diff
@@ -239,16 +270,42 @@ const ManageUsers: React.FC<Props> = ({
   useEffect(() => {
     if (!activeUser && sortedUsers.length) {
       updateActiveUser(sortedUsers[0]);
+      queryUserWithPermissions();
     }
-  }, [activeUser, sortedUsers]);
+  }, [activeUser, sortedUsers, queryUserWithPermissions]);
 
-  // Navigate to addeed user, if applicable
+  // Navigate to added user, if applicable
   useEffect(() => {
     if (addedUserId && usersState[addedUserId]) {
       updateActiveUser(usersState[addedUserId]);
+      queryUserWithPermissions();
       setAddedUserId(undefined);
     }
-  }, [addedUserId, usersState]);
+  }, [addedUserId, usersState, queryUserWithPermissions]);
+
+  // Navigate to correct user on user deletion (first sorted, unless the deleted user was first)
+  useEffect(() => {
+    if (deletedUserId) {
+      const nextUser: LimitedUser =
+        sortedUsers[0].id === deletedUserId && sortedUsers.length > 1
+          ? sortedUsers[1]
+          : sortedUsers[0];
+      updateActiveUser(nextUser);
+      queryUserWithPermissions();
+      setDeletedUserId(undefined);
+    }
+  }, [deletedUserId, usersState, sortedUsers, queryUserWithPermissions]);
+
+  // If there's no userWithPermisions, call the permissions query again
+  useEffect(() => {
+    if (userWithPermissions === undefined) {
+      queryUserWithPermissions();
+    }
+  }, [queryUserWithPermissions, userWithPermissions]);
```

Comment:
Oh thank you, that documentation is helpful! I didn't really know what those args were, I was just following the pattern elsewhere, so thanks.

Unfortunately removing `queryUserWithPermissions` here triggers a lint warning that:

> React Hook useEffect has a missing dependency: 'queryUserWithPermissions'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
> 

Do you know if this is valid or if it's something we can safely turn off the linter for?

Reply:
Oh, interesting... I think we actually want to keep the rule on. Here's an interesting [StackOverflow thread](https://stackoverflow.com/a/58959607/5365414) about the issue. There's also a page in the [React docs](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies) about it... they recommend moving the function into the hook, or outside of the component... But I'm not sure what the right move here is :thinking: 
What do you think?

Reply:
in my experience, you never want to turn the linter of for react effects!

Reply:
Hmmm this is a bit of a dilemma. I don't love the idea of moving the function into the hook, since there are multiple places outside hooks where it also needs to be used. (I also got errors before when I tried to do that, though I haven't tried with `lazyQuery`.) We could probably move it outside the component, but since it's using that `onComplete` to update the value of `userWithPermissions` I don't know what kind of errors we might run into once it no longer has access to state variables.

One example in the docs was to wrap it in a callback, but that just throws the following error again:

> React Hook "useLazyQuery" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function  react-hooks/rules-of-hooks

Is there any danger in just keeping the function as a parameter to the `useEffects`? All our other options here seem unideal.

Reply:
I guess the danger is more render cycles than we need, potentially hurting performance? But for now it's probably fine to leave as is. Sorry for the rabbit hole :confounded:

Reply:
Please don't apologize, I learned a lot from this rabbit hole! :) But unfortunately yeah, I think leaving it as is for now is probably our best bet. If we ever figure out how to bubble up the query into `ManageUserContainer` then we can definitely revisit this though.